fendermint/vm/interpreter/src/fvm/gas/default.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::fvm::gas::GasLayer;
+use crate::fvm::state::{read_actor_state, FvmExecState};
+use crate::fvm::FvmMessage;
+use anyhow::anyhow;
+use fendermint_actor_gas::{Gas, State};
+use fendermint_vm_actor_interface::gas::GAS_ACTOR_ADDR;
+use fendermint_vm_actor_interface::system;
+use fvm_ipld_blockstore::Blockstore;
+use std::marker::PhantomData;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+type AtomicGas = AtomicU64;
+
+pub struct DefaultGas<DB> {
+    /// The total gas available to be used by transactions in the current block
+    /// The executor requires Send + Sync, using an atomic variable instead of u64.
+    block_gas_quota: AtomicGas,
```

Comment:
I didnt use the `limit` here because with each txn, the available block gas deducts the gas used. So that block gas limit just needs to be read once when a new block starts.